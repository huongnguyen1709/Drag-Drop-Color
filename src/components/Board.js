import React, { Component } from 'react'
import initialData from './initialData'

import Box from './Box'

class Board extends Component {

    state = initialData

    findIndex = (boxes, id) => {
        var result = -1;
        boxes.forEach((box, index) => {
            if (box.id === id) {
                result = index;
            }
        })
        return result;
    }

    swapBox = (dragId, dropId) => {
        var { boxes } = this.state
        const fromIndex = this.findIndex(boxes, dragId)
        const toIndex = this.findIndex(boxes, dropId)
        const fromBox = boxes[fromIndex]
        const toBox = boxes[toIndex]
        boxes.splice(fromIndex, 1, toBox)
        boxes.splice(toIndex, 1, fromBox)
        this.setState({
            boxes: boxes
        })
    }

    handleDrop = e => {
        const drag_id = e.dataTransfer.getData('drag_id')
        e.preventDefault()
        const drop_id = e.target.id
        this.swapBox(drag_id, drop_id)
        const box = document.getElementById(drag_id)
        box.style.visibility = 'visible'
    }

    render() {
        return (
            <div
                className="board"
                onDrop={this.handleDrop}
            >
                {this.state.boxes.map(box => {
                    return (
                        <Box
                            key={box.id}
                            box={box}
                            draggable="true"
                        />)
                })}
            </div>
        );
    }
}

export default Board;