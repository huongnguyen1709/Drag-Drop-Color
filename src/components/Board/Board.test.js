import React from 'react';
import { shallow } from 'enzyme';
import Board from './Board';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

describe('Board', () => {
    it('should render correctly in "debug" mode', () => {
        const component = shallow(<Board debug />);

        expect(component).toMatchSnapshot();
    });
});