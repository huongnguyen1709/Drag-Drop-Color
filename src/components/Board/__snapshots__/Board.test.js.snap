// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Board should render correctly in "debug" mode 1`] = `ShallowWrapper {}`;

exports[`Board should render correctly in "debug" mode 1`] = `ShallowWrapper {}`;
